distributions/openhab/src/main/resources/bin/restore

```diff
@@ -0,0 +1,191 @@
+    exit 1
+  }
+}
+
+echo "                                          "
+echo "##########################################"
+echo "       openHAB 2.x.x restore script       "
+echo "##########################################"
+echo "                                          "
+
+InputFile="$1"
+
+setup "$InputFile"
+
+## Extract zip file
+echo "Extracting zip file to temporary folder."
+
+unzip -oq "$InputFile" -d "$TempDir" || {
+  echo "Unable to unzip $InputFile, Aborting..." >&2
+  exit 1
+}
+
+## Check for backup properties list.
+if [ ! -f "$TempDir/backup.properties" ]; then
+  echo "Backup was not created by openHAB scripts, please resort to a manual restore..." >&2
+  exit 1
+fi
+
+## Grab information with backup.properties
+str="$(awk '/version=/{print $1}' "$TempDir/backup.properties")"
+BackupVersion=${str#*=}
+str="$(awk '/timestamp=/{print $1}' "$TempDir/backup.properties")"
+BackupTime=${str#*=}
+str="$(awk '/user=/{print $1}' "$TempDir/backup.properties")"
+OHUser=${str#*=}
+str="$(awk '/group=/{print $1}' "$TempDir/backup.properties")"
+OHGroup=${str#*=}
+
+## Feeback to user
+echo ""
+echo " Backup Information:"
+echo " -------------------"
+echo " Backup Version         | $BackupVersion (You are on $CurrentVersion)"
+echo " Backup Timestamp       | $BackupTime"
+echo " Config belongs to user | $OHUser"
+echo "             from group | $OHGroup"
+echo ""
+echo "Your current configuration will become owned by $OHUser:$OHGroup."
+echo ""
+echo "Any existing files with the same name will be replaced."
+echo "Any file without a replacement will be deleted."
+echo ""
+printf "Okay to Continue? [y/N]: "
+read -r answer
+case "$answer" in
+  [Yy]*)
+    ;;
+  *)
+    echo "Cancelling restore..."
+    rm -rf /tmp/openhab2
+    exit 0
+    ;;
+esac
+
+## Move old configuration
+rm -rf /tmp/openhab/old
+mkdir -p /tmp/openhab/old
+echo "Moving system files in userdata to temporary folder"
+if [ -d "$OPENHAB_USERDATA/backups" ]; then
+  mv "$OPENHAB_USERDATA/backups" /tmp/openhab/old || {
+    echo "Could not move backup folder to temporary folder..." >&2
+    exit 1
+  }
+fi
+if [ -d "${OPENHAB_USERDATA:?}/etc" ]; then
+  mv "${OPENHAB_USERDATA:?}/etc" /tmp/openhab/old || {
+    echo "Could not move etc folder to temporary folder" >&2
+    exit 1
+  }
+fi
+
+echo "Deleting old userdata folder..."
+rm -rf "${OPENHAB_USERDATA:?}/"*
+
+echo "Restoring system files in userdata..."
+if [ -d /tmp/openhab/old/backups ]; then
+  mv /tmp/openhab/old/backups "${OPENHAB_USERDATA:?}/" || {
+    echo "Unable to move other backup files back..."
+    exit 1
+  }
+fi
+if [ -d /tmp/openhab/old/etc ]; then
+  mv /tmp/openhab/old/etc "${OPENHAB_USERDATA:?}/" || {
+    echo "Unable to move system files back..."
+    exit 1
+  }
+fi
+
+echo "Deleting old conf folder..."
+rm -rf "${OPENHAB_CONF:?}/"*
+
+
+## Restore configuration
+echo "Restoring openHAB with backup configuration..."
+command cp -af "$TempDir/conf/"*     "${OPENHAB_CONF:?}/" || {
```

Comment:
Some distributions use an alias for cp which includes the -i flag. 

https://serverfault.com/questions/119869/turning-off-cp-copy-commands-interactive-mode-cp-overwrite